src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -3183,6 +3205,8 @@ void MavlinkReceiver::update_rx_stats(const mavlink_message_t &message)
 				_component_states[i].last_time_received_ms = hrt_absolute_time() / 1000;
 				_component_states[i].last_sequence = message.seq;
 
+				++_component_states_count;
```

Comment:
What about changing this to:
```suggestion
				_component_states_count = i + 1;
```
I think that makes it slightly clearer what is going on.